src/bin/pg_autoctl/cli_common.c

```diff
@@ -406,16 +407,13 @@ cli_create_node_getopts(int argc, char **argv,
 	 */
 	if (IS_EMPTY_STRING_BUFFER(LocalOptionConfig.pgSetup.pgdata))
 	{
-		char *pgdata = getenv("PGDATA");
-
-		if (pgdata == NULL)
+		int pgdatalen = get_env_variable("PGDATA", LocalOptionConfig.pgSetup.pgdata, MAXPGPATH);
```

Comment:
We fetch PG_DATA a lot of times, all with this same error. I think it makes sense to put this in a function instead, e.g. `get_env_PGDATA_or_fatal(char** result)`

Reply:
Maybe something like `getPGDATA(char *pgdata, int size, bool error_if_not_found)`.

Reply:
get_env_pgdata, however without the error message